### PR TITLE
[INTERNAL] File System: Add option to apply gitignore excludes

### PR DESCRIPTION
**Pull request by RandomByte**
This makes use of a standard feature of globby:
https://github.com/sindresorhus/globby#gitignore

This can be usefull when working with a project's root-reader
(Specification#getRootReader) where one might expect directories like
.git and node_modules to be excluded

### CODE REVIEW

File: test/lib/adapters/FileSystem_read.js [hunk 1]
```diff
@@ -399,3 +399,37 @@ test("static excludes: glob with negated directory exclude, not excluding resour
...
+test("glob with useGitignore: true", async (t) => {
+	const srcReader = createAdapter({
+		fsBasePath: "./test/fixtures/library.l/",
+		virBasePath: "/",
+		useGitignore: true
+	});
+
+	const resources = await srcReader.byGlob("/**/*");
+
+	t.is(resources.length, 5, "Found two resources");
```
Comment (d3xter666) [inline test/lib/adapters/FileSystem_read.js:412]: two?

Comment (RandomByte) [reply test/lib/adapters/FileSystem_read.js]: Done

File: lib/adapters/FileSystem.js [hunk 2]
```diff
@@ -28,11 +29,14 @@ class FileSystem extends AbstractAdapter {
 	 * @param {string} parameters.virBasePath Virtual base path
 	 * @param {string} parameters.fsBasePath (Physical) File system path
 	 * @param {string[]} [parameters.excludes] List of glob patterns to exclude
-	 * @param {object} [parameters.project] Experimental, internal parameter. Do not use
+	 * @param {object} [parameters.useGitignore=false]
+	 *   Whether to apply any excludes defined in an optional .gitignore in the base directory
```
Comment (flovogt) [inline lib/adapters/FileSystem.js:33]: should we be here more precise by naming the base directory
```suggestion
	 *   Whether to apply any excludes defined in an optional .gitignore in the given <code>fsBasePath</code> directory
```

**Review by flovogt — APPROVED** [review]

File: lib/adapters/FileSystem.js [hunk 3]
```diff
@@ -129,65 +134,71 @@ class FileSystem extends AbstractAdapter {
...
 
-					if (!stat.isDirectory()) {
-						// Add content
-						options.createStream = function() {
-							return fs.createReadStream(fsPath);
-						};
-					}
+		if (this._useGitignore) {
+			// Check whether path should be ignored
+			const isIgnored = await isGitIgnored();
```
Comment (matz3) [inline lib/adapters/FileSystem.js:165]: I would expect that `this._fsBasePath` needs to be passed here as `cwd`.

Comment (matz3) [reply lib/adapters/FileSystem.js:165]: Follow-up PR: https://github.com/SAP/ui5-fs/pull/465

**Pull request merged**